pkg/operation/garden/garden.go

```diff
@@ -41,6 +42,22 @@ func New(projectLister gardenlisters.ProjectLister, namespace string) (*Garden,
 	}, nil
 }
 
+//Check annotation "dns.garden.sapcloud.io/hostedZoneID", return true if the field is satisfied
+func checkDNSHostedZoneID(annotations map[string]string) bool {
+	if annotations == nil {
+		return false
+	}
+
+	r := true
+	//DNSHostedZoneID is not needed for AlicloudDNS
+	skip := annotations[common.DNSProvider] == string(gardenapis.DNSAlicloud)
+	if !skip {
+		_, r = annotations[common.DNSHostedZoneID]
+	}
+
+	return r
```

Comment:
Refactor this function to improve readability:

```golang
func ... {
  if annotations == nil {
    return false
  }

  // Alicloud DNS does not support a hosted zone ID
  if dnsProvider, ok := annotations[common.DNSProvider]; ok && dnsProvider == string(garden.DNSAlicloud) {
    return true
  }

  _, ok := annotations[common.DNSHostedZoneID]
  return ok
}
```

Reply:
Thanks for your example. I will do further refactory to make it more reasonable and readable.